Composer/packages/extension-client/src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { useMemo } from 'react';
+
+const KEY = 'composer:extensions';
+
+export function useLocalStorage() {
+  const { __extensionId, __bundleId } = window.Composer;
+
+  const getItem = useMemo(
+    () => (key: string) => {
+      const allStates = window.localStorage.getItem(KEY);
+      if (allStates) {
+        const extensionState = JSON.parse(allStates)?.[__extensionId];
+        return extensionState?.[__bundleId]?.[key];
+      }
+      return undefined;
+    },
+    [__extensionId, __bundleId]
+  );
+
+  const getAll = useMemo(
+    () => () => {
+      const allStatesString = window.localStorage.getItem(KEY);
+      if (allStatesString) {
+        return JSON.parse(allStatesString)?.[__extensionId]?.[__bundleId];
+      }
+    },
+    [__extensionId, __bundleId]
+  );
+
+  const setItem = useMemo(
+    () => (key: string, value: any) => {
+      const allStatesString = window.localStorage.getItem(KEY);
+      const allStates = allStatesString ? JSON.parse(allStatesString) : {};
+
+      const extensionState = allStates?.[__extensionId] || {};
+      const bundleState = extensionState?.[__bundleId] || {};

```

Comment:
These should also be extracted into a helper method or const since it is used frequently.